peer_mentoring/chat/consumers.py

```diff
@@ -44,3 +45,51 @@ def save_chat_message(self, message, now):
 
     async def chat_message(self, event):
         await self.send(text_data=json.dumps(event))
+
+
+class PrivateChatConsumer(AsyncWebsocketConsumer):
+    async def connect(self):
+        self.sender = self.scope["user"]
+        self.receiver_id = self.scope["url_route"]["kwargs"]["receiver_id"]
+        # sort the ids to create a unique room name for the chat between these two users
+        ids = sorted([self.sender.id, int(self.receiver_id)])
+        self.room_group_name = f"private_chat_{ids[0]}_{ids[1]}"
+        await self.channel_layer.group_add(self.room_group_name, self.channel_name)
+        await self.accept()
+
+    async def disconnect(self, close_code):
+        await self.channel_layer.group_discard(self.room_group_name, self.channel_name)
+
+    async def receive(self, text_data):
+        text_data_json = json.loads(text_data)
+        message = text_data_json["message"]
+        now = timezone.now()
+
+        # only allow messages from the two users in the chat
+        if self.sender.id == int(self.receiver_id) or self.sender.id == int(
```

Comment:
should we move this check up to the writing to db part? to guarantee we only get the messages of both users written to the db? then you might not need this check anymore